### PR TITLE
Added schaake_partition, utility and unit test

### PR DESCRIPTION
**Pull request by stcui007**
Added schaake_partition, utility and unit test

## Additions
modified:   models/kernels/schaake_partitioning.hpp
modified:   models/kernels/utility.hpp
new file:   test/models/hymod/include/Schaake_Partitioning_Test.cpp
-

## Removals

-

## Changes

-

## Testing

Added unit test and passed

## Notes

-

## Todos

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows project standards (link if applicable)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

## Testing checklist (automated report can be put here)

1.

### Target Environment support

- [ ] Linux

### CODE REVIEW

**Comment by hellkite500** [comment]
Looking at the changes, I'm confused as to what is different from #52.  Seems like some of the same changes appear here...

**Comment by stcui007** [comment]
There is no change to Pdm03.h. I just removed it from the pull request.

File: CMakeFiles/cmake.check_cache [hunk 1]
```diff
@@ -0,0 +1 @@
+# This file is generated by cmake for dependency checking of the CMakeCache.txt file
```
Comment (hellkite500) [inline CMakeFiles/cmake.check_cache:1]: Please remove from the index, generated files should be ignored.

File: CMakeCache.txt [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# This is the CMakeCache file.
```
Comment (hellkite500) [inline CMakeCache.txt:1]: Remove from index

**Comment by hellkite500** [comment]
Replaced by #102

**Pull request closed without merging**